### PR TITLE
Add new `Style/UseNamedBlockArgumentForSimpleMethodCalls` cop

### PR DESCRIPTION
**Pull request by dabroz**
Add new `Style/UseNamedBlockArgumentForSimpleMethodCalls` cop

I wanted `array.map { _1.name }` to be corrected to `array.map(&:name)` but instead of adding more stuff to `SymbolProc` it seemed better to add a new cop instead that could be later picked up with `SymbolProc`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by dabroz** [comment]
`RuboCop::Cop::Style::RedundantRegexpEscape with multibyte characters removes the escape character at the right position` seems to be broken on `master`.

**Comment by marcandre** [comment]
I think this is a promising cop, but it should really go both ways (e.g. if there's only a single param, and the block is single line, then use `_1` instead).

Independently, I think `SymbolProc` should be fixed to catch `{ _1.foo }`

**Comment by bbatsov** [comment]
Totally agree with @marcandre.

File: config/default.yml [hunk 1]
```diff
@@ -4554,6 +4554,11 @@ Style/UnpackFirst:
   Enabled: true
   VersionAdded: '0.54'
 
+Style/UseNamedBlockArgumentForSimpleMethodCalls:
+  Description: "{ _1.method } should not be used because it won't be picked up by SymbolProc cop."
```
Comment (bbatsov) [inline config/default.yml:4558]: Btw, that's no longer true, after a recent change. :-)

**Comment by bbatsov** [comment]
It seems to me a better rationale for this cop would be to stick to their explicit or implicit params, with the option to apply this only to one-line blocks or to all blocks.

**Comment by bbatsov** [comment]
I think @dvandersluis already fixed this one in `SymbolProc`.

**Pull request closed without merging**